src/timeout.c

```diff
@@ -164,12 +164,19 @@ void handleBlockedClientsTimeout(void) {
 int getTimeoutFromObjectOrReply(client *c, robj *object, mstime_t *timeout, int unit) {
     long long tval;
     long double ftval;
+    mstime_t now = mstime();
```

Comment:
we can revert this change, and then maybe we won't have conflicts with #10300

Reply:
Logically it is incorrect to use mstime() instead of `now` because the condition might be true and later on the addition (`tval += now`) will be overflow. Prefer to leave it as is.

Reply:
i wrote that because in the previous version of your PR there was only one reference to `now`, so it was identical.
but now that we avoid UB, we have several, so we'll just need to resolve conflicts later...